yoga/Yoga.cpp

```diff
@@ -106,6 +105,51 @@ static int YGDefaultLog(
 #undef YG_UNUSED
 #endif
 
+#ifdef _MSC_VER
+#define YG_ALLOC_ALIGNED ([](size_t a, size_t s) -> void* { \
+  return _aligned_malloc(s, a);                             \
+})
+#define YG_FREE_ALIGNED ([](void* m) -> void {              \
+  _aligned_free(m);                                         \
+})
+#else
+#define YG_ALLOC_ALIGNED ([](size_t a, size_t s) -> void* { \
+  return aligned_alloc(a, s);                               \
+})
+#define YG_FREE_ALIGNED ([](void* m) -> void {              \
+  free(m);                                                  \
+})
+#endif
+
+static YGAllocatorAllocateFunc gAllocatorAllocateFunc = YG_ALLOC_ALIGNED;
+static YGAllocatorFreeFunc gAllocatorFreeFunc = YG_FREE_ALIGNED;
```

Comment:
If these are global, they will need to be protected by mutex, in case multiple threads are using Yoga at once.

I think it would be cleaner if the allocator was set as part of YGConfig, though it would mean YGConfig itself could not be allocated using the custom allocator. Though, Yoga may otherwise call functions which perform their dynamic allocations, so if the goal is just for the large structures, I think it could still be okay.